cardinal/ecs/component.go

```diff
@@ -115,6 +115,26 @@ func (c *ComponentType[T]) MustFirst(w *World) entity.ID {
 	return id
 }
 
+func CreateMany(world *World, num int, components ...component.IAbstractComponent) ([]entity.ID, error) {
+	acc := make([]IComponentType, 0, len(components))
+	for _, comp := range components {
+		c, ok := world.GetComponentByName(comp.Name())
```

Comment:
This pattern of calling GetComponentByName, and returning an error when it's absent seems to be repeated. I suggest making GetComponentByName return an error, and assigning the "Must register component before creating an entity" to an actual error variable.